tests/e2e/specs/api/PhpDevFileAPI.spec.ts

```diff
@@ -0,0 +1,70 @@
+/** *******************************************************************
+ * copyright (c) 2023 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import { BASE_TEST_CONSTANTS } from '../../constants/BASE_TEST_CONSTANTS';
+import { e2eContainer } from '../../configs/inversify.config';
+import { CLASSES } from '../../configs/inversify.types';
+import {Logger} from "../../utils/Logger";
+import {DevfilesRegistryHelper} from "../../utils/DevfilesRegistryHelper";
+import {ContainerTerminal, KubernetesCommandLineToolsExecutor} from "../../utils/KubernetesCommandLineToolsExecutor";
+import {ShellExecutor} from "../../utils/ShellExecutor";
+import {DevWorkspaceConfigurationHelper} from "../../utils/DevWorkspaceConfigurationHelper";
+import {DevfileContext} from "@eclipse-che/che-devworkspace-generator/lib/api/devfile-context";
+import {ShellString} from "shelljs";
+import {expect} from "chai";
+import {API_TEST_CONSTANTS} from "../../constants/API_TEST_CONSTANTS";
+import YAML from "yaml";
+import {StringUtil} from "../../utils/StringUtil";
+import {string} from "yaml/dist/schema/common/string";
+
+
+suite('PHP devfile API test', function (): void {
+    const devfilesRegistryHelper: DevfilesRegistryHelper = e2eContainer.get(CLASSES.DevfilesRegistryHelper);
+    const kubernetesCommandLineToolsExecutor: KubernetesCommandLineToolsExecutor = e2eContainer.get(CLASSES.KubernetesCommandLineToolsExecutor);
+    let containerTerminal: ContainerTerminal = e2eContainer.get(CLASSES.ContainerTerminal);
+    let devWorkspaceConfigurationHelper: DevWorkspaceConfigurationHelper;
+    let devfileContext: DevfileContext;
+    let devfileContent: string ='';
+
+    suiteSetup(`Prepare login ${BASE_TEST_CONSTANTS.TEST_ENVIRONMENT}`, async function (): Promise<void> {
+        kubernetesCommandLineToolsExecutor.loginToOcp();
+    });
+
+    test('Create  PHP DevSpace', async function (): Promise<void> {
+        kubernetesCommandLineToolsExecutor.namespace = 'admin-devspaces';
+        devfileContent = devfilesRegistryHelper.getDevfileContent('php');
```

Comment:
'php' would better to extract as a DEVFILE_ID constant to make localizing the sample name simpler when investigate the test failure.
You can extract devfile ids of Dev Spaces 3.17 from the next json obtained using request `curl -X GET http://localhost:8081/dashboard/api/airgap-sample -H 'accept: */*' -H 'Authorization: Bearer ...:
```
 [INFO] Devfiles metadata content:
 [
   {
     "id": "php",
     "displayName": "PHP",
     "description": "PHP Stack with PHP",
     "tags": [
       "Tech-Preview",
       "PHP",
       "UBI8"
     ],
     "url": "http://devspaces-dashboard.openshift-devspaces.svc:8080/dashboard/api/airgap-sample/devfile/download?id=php",
     "icon": {
       "mediatype": "image/svg+xml"
     },
     "project": {
       "zip": {
         "filename": "devspaces-samples-php-hello-world-devspaces-3.16-rhel-8.zip"
       }
     },
     "devfile": {
       "filename": "devspaces-samples-php-hello-world-devspaces-3.16-rhel-8-devfile.yaml"
     }
   },
   {
     "id": "dotnet",
     "displayName": ".NET",
     "description": ".NET stack with .NET Core SDK 8, Runtime, C# Language Support and Debugger",
     "tags": [
       "Tech-Preview",
       ".NET",
       "C#",
       ".NET SDK",
       ".NET Runtime",
       "Netcoredbg",
       "Omnisharp",
       "UBI8"
     ],
     "url": "http://devspaces-dashboard.openshift-devspaces.svc:8080/dashboard/api/airgap-sample/devfile/download?id=dotnet",
     "icon": {
       "mediatype": "image/svg+xml"
     },
     "project": {
       "zip": {
         "filename": "devspaces-samples-dotnet-web-simple-devspaces-3.16-rhel-8.zip"
       }
     },
     "devfile": {
       "filename": "devspaces-samples-dotnet-web-simple-devspaces-3.16-rhel-8-devfile.yaml"
     }
   },
   {
     "id": "nodejs-mongodb",
     "displayName": "Node.js MongoDB",
     "description": "Node.js stack with NPM 8, Node.js 16 and MongoDB 3.6",
     "tags": [
       "Node.js",
       "NPM",
       "Express",
       "MongoDB",
       "UBI8"
     ],
     "url": "http://devspaces-dashboard.openshift-devspaces.svc:8080/dashboard/api/airgap-sample/devfile/download?id=nodejs-mongodb",
     "icon": {
       "mediatype": "image/svg+xml"
     },
     "project": {
       "zip": {
         "filename": "devspaces-samples-nodejs-mongodb-sample-devspaces-3.16-rhel-8.zip"
       }
     },
     "devfile": {
       "filename": "devspaces-samples-nodejs-mongodb-sample-devspaces-3.16-rhel-8-devfile.yaml"
     }
   },
   {
     "id": "java-lombok",
     "displayName": "Java Lombok",
     "description": "Java Stack with Project Lombok",
     "tags": [
       "Java",
       "OpenJDK",
       "Maven",
       "Lombok",
       "UBI8"
     ],
     "url": "http://devspaces-dashboard.openshift-devspaces.svc:8080/dashboard/api/airgap-sample/devfile/download?id=java-lombok",
     "icon": {
       "mediatype": "image/svg+xml"
     },
     "project": {
       "zip": {
         "filename": "devspaces-samples-lombok-project-sample-devspaces-3.16-rhel-8.zip"
       }
     },
     "devfile": {
       "filename": "devspaces-samples-lombok-project-sample-devspaces-3.16-rhel-8-devfile.yaml"
     }
   },
   {
     "id": "nodejs-express",
     "displayName": "Node.js Express",
     "description": "Node.js stack with NPM 8, Node.js 16 and Express Web Application",
     "tags": [
       "Node.js",
       "NPM",
       "Express",
       "UBI8"
     ],
     "url": "http://devspaces-dashboard.openshift-devspaces.svc:8080/dashboard/api/airgap-sample/devfile/download?id=nodejs-express",
     "icon": {
       "mediatype": "image/svg+xml"
     },
     "project": {
       "zip": {
         "filename": "devspaces-samples-web-nodejs-sample-devspaces-3.16-rhel-8.zip"
       }
     },
     "devfile": {
       "filename": "devspaces-samples-web-nodejs-sample-devspaces-3.16-rhel-8-devfile.yaml"
     }
   },
   {
     "id": "cpp",
     "displayName": "C/C++",
     "description": "C and C++ Developer Tools stack with GCC, cmake and make",
     "tags": [
       "Tech-Preview",
       "C",
       "C++",
       "Clang",
       "GCC",
       "g++",
       "make",
       "cmake",
       "UBI8"
     ],
     "url": "http://devspaces-dashboard.openshift-devspaces.svc:8080/dashboard/api/airgap-sample/devfile/download?id=cpp",
     "icon": {
       "mediatype": "image/svg+xml"
     },
     "project": {
       "zip": {
         "filename": "devspaces-samples-c-plus-plus-devspaces-3.16-rhel-8.zip"
       }
     },
     "devfile": {
       "filename": "devspaces-samples-c-plus-plus-devspaces-3.16-rhel-8-devfile.yaml"
     }
   },
   {
     "id": "ansible",
     "displayName": "Ansible",
     "description": "Development environment for Ansible playbook creation, testing with Molecule, and ansible-lint checks",
     "tags": [
       "Community",
       "Ansible",
       "Molecule"
     ],
     "url": "http://devspaces-dashboard.openshift-devspaces.svc:8080/dashboard/api/airgap-sample/devfile/download?id=ansible",
     "icon": {
       "mediatype": "image/svg+xml"
     },
     "project": {
       "zip": {
         "filename": "devspaces-samples-ansible-devspaces-demo-devspaces-3.16-rhel-8.zip"
       }
     },
     "devfile": {
       "filename": "devspaces-samples-ansible-devspaces-demo-devspaces-3.16-rhel-8-devfile.yaml"
     }
   },
   {
     "id": "quarkus-rest-api",
     "displayName": "Quarkus REST API",
     "description": "Quarkus stack with a default REST endpoint application sample",
     "tags": [
       "Java",
       "OpenJDK",
       "Maven",
       "Quarkus",
       "UBI8"
     ],
     "url": "http://devspaces-dashboard.openshift-devspaces.svc:8080/dashboard/api/airgap-sample/devfile/download?id=quarkus-rest-api",
     "icon": {
       "mediatype": "image/svg+xml"
     },
     "project": {
       "zip": {
         "filename": "devspaces-samples-quarkus-quickstarts-devspaces-3.16-rhel-8.zip"
       }
     },
     "devfile": {
       "filename": "devspaces-samples-quarkus-quickstarts-devspaces-3.16-rhel-8-devfile.yaml"
     }
   },
   {
     "id": "go",
     "displayName": "Go",
     "description": "Stack with Go",
     "tags": [
       "Tech-Preview",
       "Go",
       "Golang",
       "UBI8"
     ],
     "url": "http://devspaces-dashboard.openshift-devspaces.svc:8080/dashboard/api/airgap-sample/devfile/download?id=go",
     "icon": {
       "mediatype": "image/svg+xml"
     },
     "project": {
       "zip": {
         "filename": "devspaces-samples-golang-health-check-devspaces-3.16-rhel-8.zip"
       }
     },
     "devfile": {
       "filename": "devspaces-samples-golang-health-check-devspaces-3.16-rhel-8-devfile.yaml"
     }
   },
   {
     "id": "python",
     "displayName": "Python",
     "description": "Python Stack with Python 3.9 and pip",
     "tags": [
       "Python",
       "pip",
       "UBI8"
     ],
     "url": "http://devspaces-dashboard.openshift-devspaces.svc:8080/dashboard/api/airgap-sample/devfile/download?id=python",
     "icon": {
       "mediatype": "image/svg+xml"
     },
     "project": {
       "zip": {
         "filename": "devspaces-samples-python-hello-world-devspaces-3.16-rhel-8.zip"
       }
     },
     "devfile": {
       "filename": "devspaces-samples-python-hello-world-devspaces-3.16-rhel-8-devfile.yaml"
     }
   }
 ]
```